src/modifyDate/modifyDate.ts

```diff
@@ -1,20 +1,47 @@
-import moment from "moment";
+/** The current supported ranges of modifying a date */
+export type ModifyDateRange = "month" | "day" | "year" | "hours" | "seconds";
 
 /**
  * Modifies a date by either adding or substracting a certain range
  * @param {Date} from {date} the starting date
- * @param {number} value {number} the value you want added or substractedf
+ * @param {number} value {number} the value (amount) you want added or substracted
  * @param {"ADD" | "SUBTRACT"} type {ADD | SUBSTRACT} the operation type
- * @param {moment.DurationInputArg2} range {moment.DurationInputArg2} the range example , months , seconds, h, m, y etc
- * @returns {string} The modified date
+ * @param {"month" | "day" | "year" | "hours" | "seconds"} range the name of the range to modify, example: "month"
  */
-export function modifyDate(from: Date, value: number, type: "ADD" | "SUBTRACT", range: moment.DurationInputArg2): Date {
-    if (!moment(from).isValid()) {
-        from = moment().toDate();
-    }
-    switch (type) {
-        case "ADD": return moment(from).add(value, range).toDate();
-        case "SUBTRACT": return moment(from).subtract(value, range).toDate();
-        default: return moment().toDate();
+export function modifyDate(from: Date, value: number, type: "ADD" | "SUBTRACT", range: ModifyDateRange): Date {
```

Comment:
Because functions are objects in Javascript, you can declare local functions `add` and `subtract` and let the function return `this` to that you can utilize it. These functions would use closures to access the `from` date

Reply:
this is similar to how jest uses `expect`
```js
expect(something).not.toBeTruthy();
```

Reply:
I think your idea is much better and I agree with it but If we are changing the api I want to change it completely for all date utils as part of another PR. 

This PR addresses only STEP 1 which is removing moment while trying to keep the existing api as similar as possible.

So what you're describing would be STEP 2.

This way I'm making incremental changes and avoiding making too many changes as part of one PR.

Reply:
I'm adding a test case for your example

Reply:
agree